### DISCUSSION TITLE
Chakra UI styles not working in customElements

### DISCUSSION THREAD

**Discussion by amit08255** [chakra-ui/chakra-ui#8284] [Q&A] [answered]
![image](https://github.com/chakra-ui/chakra-ui/assets/28493237/fe9d8970-6e4f-4768-9fa2-7fd140efe912)

As per screenshot above, both buttons are from Chakra UI. The form component is web-component using `customElements` is under `shadowDOM` but the Chakra UI style is not being applied to this button.

```tsx
<ChakraProvider theme={theme}>
            <Button colorScheme="primary">Click me</Button>
            {
                <standalone-component
                        ref={ref}
                        name={name}
                        email={email}
                    />
            }
</ChakraProvider>
```

The standalone web-component `.js` file is hosted on Nginx.

```tsx
<form onSubmit={onSubmit}>
    <label>
        Name:
        <input type="text" name="username" value={name} onChange={onNameChange} />
    </label>
    <label>
        Email:
        <input type="text" name="email" value={email} onChange={onEmailChange} />
    </label>
    <Button type="submit" colorScheme="primary">Submit</Button>
</form>
```

**Comment by segunadebayo** [comment, marked as answer]
This is a v2 + Shadow DOM cache issue. Emotion must target the shadow root.

v2 is maintenance-only. In v3: https://chakra-ui.com/docs/get-started/environments/shadow-dom
